system/lib/libc/musl/src/unistd/isatty.c

```diff
@@ -1,9 +1,30 @@
+#ifdef __EMSCRIPTEN__
+#include <errno.h>
+#endif
 #include <unistd.h>
 #include <sys/ioctl.h>
 #include "syscall.h"
 
 int isatty(int fd)
 {
+#ifdef __EMSCRIPTEN__
+	__wasi_fdstat_t statbuf;
+	int err = __wasi_fd_fdstat_get(fd, &statbuf);
+	if (err != 0) {
+		errno = err;
+		return 0;
+	}
+
+	// All character devices are terminals (other things a Linux system would
+	// assume is a character device, like the mouse, we have special APIs for).
```

Comment:
I'm not sure about this, I mean presumably other runtimes (wasmer/wasmtime) might set__WASI_FILETYPE_CHARACTER_DEVICE for actual non-termianl char devices.  For now its fine, but it doesn't very robust long term.

Reply:
Yeah, if more character devices are added, we can revisit this.